dm/worker/server_test.go

```diff
@@ -731,3 +732,35 @@ func loadSourceConfigWithoutPassword(c *C) *config.SourceConfig {
 	sourceCfg.From.Password = "" // no password set
 	return sourceCfg
 }
+
+func (t *testServer) TestServerDataRace(c *C) {
+	var (
+		masterAddr   = tempurl.Alloc()[len("http://"):]
+		keepAliveTTL = int64(1)
+	)
+	etcdDir := c.MkDir()
+	ETCD, err := createMockETCD(etcdDir, "http://"+masterAddr)
+	c.Assert(err, IsNil)
+	defer ETCD.Close()
+	cfg := NewConfig()
+	c.Assert(cfg.Parse([]string{"-config=./dm-worker.toml"}), IsNil)
+	cfg.Join = masterAddr
+	cfg.KeepAliveTTL = keepAliveTTL
+	cfg.RelayKeepAliveTTL = keepAliveTTL
+
+	s := NewServer(cfg)
+	defer s.Close()
+	go func() {
+		err1 := s.Start()
+		c.Assert(err1, IsNil)
```

Comment:
it should return `terror.ErrWorkerServerClosed` as the replacement for data race.

Please write an unit test that can discover data race for old code, and check now data race is fixed